src/store/app.ts

```diff
@@ -27,30 +39,91 @@ export const useAppStore = defineStore('app', {
          * If the user is logged in,
          * this action gets the current user and
          * then stores its ID in store.currentUserId.
+         * Additionally, this method gets the user roles of
+         * the logged in user and saves them in the store.
+         * The 'highest' user role becomes the active user role.
          */
-        async initLogin() {
+        async initLogin(logToken?: boolean) {
             const keycloak = new Keycloak({
                 url: '/keycloak',
                 realm: 'Misarch',
                 clientId: 'frontend',
             })
 
             try {
-                await keycloak.init({
+                const authenticated = await keycloak.init({
                     onLoad: 'check-sso',
                     silentCheckSsoRedirectUri: silentCheckSsoHtmlUrl,
                     redirectUri: '/',
                 })
+
                 this.keycloak = keycloak
-                this.isLoggedIn = keycloak.authenticated ?? false
-                if (this.isLoggedIn) {
-                    const currentUser = await useClient().getCurrentUser()
-                    this.currentUserId = currentUser.currentUser?.id
+                this.isLoggedIn = authenticated
+
+                if (logToken === true) {
+                    console.log('Token', keycloak.token)
                 }
-                console.log('Token', keycloak.token)
             } catch (error) {
                 console.error('Failed to initialize adapter:', error)
             }
+
+            if (this.isLoggedIn) {
+                try {
+                    const currentUser = await useClient().getCurrentUser()
+                    this.currentUserId = currentUser.currentUser?.id
+                } catch (error) {
+                    console.error(
+                        'Failed to get the user and determine their user ID:',
+                        error
+                    )
+                }
+
+                try {
+                    const keycloakRealmAccess = keycloak.realmAccess
+                    if (
+                        keycloakRealmAccess === null ||
+                        keycloakRealmAccess === undefined
+                    ) {
+                        throw new Error(
+                            'The property realmAccess of the initialized Keycloak instance is either null or undefined.'
+                        )
+                    } else {
+                        const roles = keycloak.realmAccess?.roles
+                        if (roles === undefined) {
+                            return
+                        } else {
+                            const additionalUserRoles: UserRole[] = []
+                            if (
```

Comment:
looks a bit duplicated, maybe deduplicate with a for loop?

Reply:
I will consider doing so.